storage/local/persistence.go

```diff
@@ -650,23 +656,23 @@ func (p *persistence) checkpointSeriesMapAndHeads(fingerprintToSeries *seriesMap
 			m.series.dirty = false
```

Comment:
@beorn7 There's another problem here: we already set `dirty = false` before we have verified that the whole checkpoint was written out properly (flush, sync, close, rename). We should fix that too, but probably in a different PR then.